File: android/app/src/main/java/io/storj/mobile/storjlibmodule/models/SettingsModel.java
package io.storj.mobile.storjlibmodule.models;

import com.google.gson.annotations.Expose;
import com.google.gson.annotations.SerializedName;

import io.storj.mobile.storjlibmodule.dataprovider.contracts.SettingsContract;

/**
 * Created by Yaroslav-Note on 3/19/2018.
 */

public class SettingsModel {
    @Expose
    @SerializedName(SettingsContract._SETTINGS_ID)
    private String _id;
    @Expose
    @SerializedName(SettingsContract._FIRST_SIGN_IN)
    private boolean _isFirstSignIn;
    @Expose
    @SerializedName(SettingsContract._SYNC_STATUS)
    private boolean _syncStatus;
    @Expose
    @SerializedName(SettingsContract._SYNC_SETTINGS)
    private int _syncSettings;
    @Expose
    @SerializedName(SettingsContract._LAST_SYNC)
    private String _lastSync;

    public SettingsModel() {

    }

    public SettingsModel(String settingsId) {
        this(settingsId, false, false, 0,null);
    }

    public SettingsModel(String settingsId, boolean isFirstSignIn, boolean syncStatus, int syncSettings, String lastSync) {
        _id = settingsId;
        _syncStatus = syncStatus;
        _syncSettings = syncSettings;
        _lastSync = lastSync;
        _isFirstSignIn = isFirstSignIn;
    }

    public String getId() { return _id; }
    public boolean isFirstSignIn() { return _isFirstSignIn; }
    public boolean syncStatus() { return _syncStatus; }
    public int getSyncSettings() { return _syncSettings; }
    public String lastSync() { return _lastSync; }
}

File: ios/model/SettingsModel.m
//
//  SettingsModel.m
//  StorjMobile
//
//  Created by Bohdan Artemenko on 5/31/18.
//  Copyright © 2018 Storj. All rights reserved.
//

#import "SettingsModel.h"
#import "DictionaryUtils.h"
#import "SettingsContract.h"

@implementation SettingsModel

-(instancetype) init
{
  return [super init];
}

-(instancetype) initWithSettingsId: (NSString *) settingsId
{
  return [self initWithSettingsId:settingsId
                      firstSignIn:NO
                       syncStatus:NO
                     syncSettings:0
                         lastSync:nil];
}

-(instancetype) initWithSettingsId: (NSString *) settingsId
                       firstSignIn: (BOOL) isFirstSignIn
                        syncStatus: (BOOL) syncStatus
                      syncSettings: (int) syncSettings
                          lastSync: (NSString *) lastSync
{
  if(self = [super init])
  {
    __id = settingsId;
    _isFirstSignIn = isFirstSignIn;
    _syncStatus = syncStatus;
    _syncSettings = syncSettings;
    _lastSync = lastSync;
  }
  
  return self;
}

- (NSDictionary *)toDictionary
{
  NSMutableDictionary *dictionary = [NSMutableDictionary dictionaryWithCapacity: 5];
  [dictionary setObject: [DictionaryUtils checkAndReturnNSString: __id]
                 forKey: SettingsContract.SETTINGS_ID];
  [dictionary setObject: @(_isFirstSignIn)
                 forKey: SettingsContract.FIRST_SIGN_IN];
  [dictionary setObject: @(_syncStatus)
                 forKey: SettingsContract.SYNC_STATUS];
  [dictionary setObject: @(_syncSettings)
                 forKey: SettingsContract.SYNC_SETTINGS];
  [dictionary setObject: [DictionaryUtils checkAndReturnNSString:_lastSync]
                 forKey: SettingsContract.LAST_SYNC];
  
  return dictionary;
}

@end
